packages/web/src/store/application/ui/buy-audio/sagas.ts

```diff
@@ -371,87 +423,388 @@ Total: ${estimatedLamports.toNumber() / LAMPORTS_PER_SOL} SOL ($${
   }
 }
 
-type PopulateAndSaveTransactionDetailsArgs = {
-  purchaseTransactionId: string
-  setupTransactionId?: string
-  swapTransactionId: string
-  cleanupTransactionId?: string
-  transferTransactionId: string
-  estimatedUSD: string
-  purchasedLamports: BN
-  purchasedAudioWei: BNWei
-}
-function* populateAndSaveTransactionDetails({
-  purchaseTransactionId,
-  setupTransactionId,
-  swapTransactionId,
-  cleanupTransactionId,
-  transferTransactionId,
-  estimatedUSD,
-  purchasedLamports,
-  purchasedAudioWei
-}: PopulateAndSaveTransactionDetailsArgs) {
+function* populateAndSaveTransactionDetails() {
+  // Get transaction details from local storage
+  const localStorage = yield* getContext('localStorage')
+  const localStorageState: BuyAudioLocalStorageState =
+    (yield* call(
+      [localStorage, localStorage.getJSONValue],
+      BUY_AUDIO_LOCAL_STORAGE_KEY
+    )) ?? defaultBuyAudioLocalStorageState
+  const {
+    purchaseTransactionId,
+    setupTransactionId,
+    swapTransactionId,
+    cleanupTransactionId,
+    transferTransactionId,
+    estimatedUSD,
+    purchasedLamports,
+    purchasedAudioWei
+  } = localStorageState.transactionDetailsArgs
+
+  if (!transferTransactionId) {
+    throw new Error('Missing transactionDetailsArgs[transferTransactionId]')
```

Comment:
Nit: if this happens maybe we should zero out local storage so the next session is fine